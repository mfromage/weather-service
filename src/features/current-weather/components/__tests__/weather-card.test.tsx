import { render } from '@testing-library/react-native';
import React from 'react';
import WeatherCard from '../weather-card';

describe('WeatherCard', () => {
  const setup = () =>
    render(<WeatherCard name="Frankfurt am Main" temp={5.15} summary="Mist" />);

  it('renders correctly with given props', () => {
    const { getByText } = setup();

    expect(getByText('Frankfurt am Main')).toBeTruthy();
    expect(getByText('5.15ยบ')).toBeTruthy();
    expect(getByText('Mist')).toBeTruthy();
  });
  it('matches the snapshot', () => {
    const { toJSON } = setup();
    expect(toJSON()).toMatchSnapshot();
  });
});
