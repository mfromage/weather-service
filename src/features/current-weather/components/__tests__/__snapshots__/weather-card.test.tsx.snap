// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WeatherCard matches the snapshot 1`] = `
<View
  style={
    {
      "backgroundColor": "lightgray",
      "borderRadius": 16,
      "gap": 16,
      "padding": 16,
    }
  }
>
  <View
    style={
      {
        "flexDirection": "row",
        "justifyContent": "space-between",
      }
    }
  >
    <Text
      style={
        {
          "fontSize": 20,
          "fontWeight": "bold",
        }
      }
    >
      Frankfurt am Main
    </Text>
    <Text
      style={
        {
          "fontSize": 24,
        }
      }
    >
      5.15
      ยบ
    </Text>
  </View>
  <Text>
    Mist
  </Text>
</View>
`;
